Rebalance scanning times

StaffiX opened the pull request:
When you first make your assembly line, you have to scan the items you want to craft via the scanner. Because of this and LuV
components having a 2 hour long base recipe times, this ends up with an impossible to skip time gate of 450 seconds per LuV components.

To avoid this unnecessary waiting, this pr changes every assembly line scanning time based on this spreadsheet:
https://docs.google.com/spreadsheets/d/1a5RYvzU5wojFnCEUCVY6xqP79Cp8jW6Ce48WYzqv2nA/

This pr also modifies the base voltage of the scanner recipes one voltage below the best available scanner for the recipe, to avoid very long base recipe times, while also not forcing a scanner upgrade in every tier.

The idea is to keep scanner recipes between 15 and 90 seconds when using the best available scanner tier for the recipe.

Since duration never exceeds 180 seconds, this also changes the Research Station scanner mode to also multiply it by base voltage, and divide it by 30 so higher tier items would need more computation.

Before:
![image](https://github.com/user-attachments/assets/17f4ea3f-8ffa-414b-b74b-bb8b2f300728)
![image](https://github.com/user-attachments/assets/404b4a16-509d-41fd-a043-aa8ad85ef43c)
![image](https://github.com/user-attachments/assets/e2051d83-40e2-45cf-b9c9-06e1bfe37274)

After:
![image](https://github.com/user-attachments/assets/00b081d9-64a6-4adc-86ea-0223705934df)
![image](https://github.com/user-attachments/assets/d9f74c7b-14d9-462e-b459-a6c0c082f05b)
![image](https://github.com/user-attachments/assets/cdb77f51-c57f-4171-8e01-eb7d84f2553f)

Added the new computation formula for the research station:
![image](https://github.com/user-attachments/assets/017d79b7-09f4-48c9-8301-463249246992)

Dream-Master commented:
This is major balance change and need to be discussed .

Ruling-0 commented:
It's large in scale but I wouldn't call it a significant balance change. Scanning is almost entirely self-contained, and every single one is a once-off.

Game design wise, I see two purposes to time gates. The first is to add value to something (an example being a large craft like God Forge, which takes significant time but that is proportional to the scale/utility and adds to payoff) or to direct a player to progress/optimize (this is more reflective of recipe times, such as smelting, which constantly incentivizes the player to upgrade smelting/cooling infra).

These fit neither of the above categories. They are once-off, inconsequential crafts that only serve to hard-gate any subsequent progression, and so the time is a point of annoyance and not adding value since it's so disproportional to the actual result. Further, the absolute minimum time to scan each LuV component is 450 seconds - there is no optimizing or infra upgrading that can be done to reduce that.

This was discussed briefly in the discord, but what is kept is the soft incentive to upgrade your scanner to the most up-to-date one, by introducing perfect OC to it. This encourages the same progression in a far more elegant and less annoying fashion.

chochem commented:
a singleblock GT machine with a 4/4 OC is a strange oddity. Arent there more elegant solutions to the same problem? E.g., you could give the recipes proper voltages and then reduce the times a bit. That would also do what you hope to achieve, right?

miaowwwwww commented:
I like the idea and the way it is done. though 44 oc for singleblock is only available to UU, i think it is ok to be implemented here as well.

Ruling-0 commented:
> a singleblock GT machine with a 4/4 OC is a strange oddity. Arent there more elegant solutions to the same problem? E.g., you could give the recipes proper voltages and then reduce the times a bit. That would also do what you hope to achieve, right?

It's down to whether it should be a soft incentive to upgrade (which 4/4 OC does well) or a hard requirement. I think the soft incentive is more interesting, and this having 4/4 OC should be fine as long as the tooltip specifies the machine gets perfect OCs (which should be universal, including onto the UUM stuff that does have it undocumented).

chochem commented:
its obviously a gradual slope but every special rule and oddity makes the pack more convoluted, so the question is if there is a sufficiently strong reason to do so.
For uum the strong reason is the constant uum to eu ratio that has a 10 year history.

Dream-Master commented:
remove perfect oc and change recipe time if needed and i am fine with this pr

boubou19 commented:
yeah base issues are recipe times, OC is the lazy solution to avoid changing the recipes but the proper solution is to reduce the recipe time and increase voltage. You don't need to OC if you have a proper recipe voltage and duration.

StaffiX commented:
Alright i reverted the perfect oc, and changed scanning times.
I kept t-1 (from t-2) scanner gating to not hard require a scanner upgrade in every tier.

HoleFish commented:
It was easy to explain the required computation of research station in scanner mode (base recipe ticks). Now that it's changed, a documentation is needed to clarify computation calculation.

serenibyss commented on src/main/java/bartworks/common/loaders/recipes/AssemblyLine.java:
```diff
@@ -27,9 +28,10 @@ public void run() {
         GTValues.RA.stdBuilder()
             .metadata(RESEARCH_ITEM, ItemList.OreDrill4.get(1L))
-            .metadata(RESEARCH_TIME, 25 * MINUTES + 36 * SECONDS)
+            .metadata(SCANNING, new Scanning(16 * MINUTES, TierEU.RECIPE_IV))
```
This doesn't seem to match the spreadsheet

serenibyss commented on src/main/java/gregtech/api/util/recipe/Scanning.java:
```diff
@@ -0,0 +1,12 @@
+package gregtech.api.util.recipe;
+
+public class Scanning {
```
I would implement some basic methods in this class just in case (`hashcode`, `equals`, `toString`)

Dream-Master commented:
will test it today and compare recipes
